components/CostsChart.client.vue

```diff
@@ -0,0 +1,222 @@
+<template>
+  <div
+    id="costsChartContainer"
+    ref="chartContainer"
+    :data-summary="JSON.stringify(seriesData)"
+  />
+</template>
+
+<script lang="ts" setup>
+import * as Highcharts from "highcharts";
+import "highcharts/modules/accessibility";
+import "highcharts/modules/exporting";
+import "highcharts/modules/export-data";
+import "highcharts/modules/offline-exporting";
+
+import throttle from "lodash.throttle";
+import { chartBackgroundColorOnExporting, chartOptions, colorBlindSafeColors, contextButtonOptions, costsChartLabelFormatter, costsChartStackLabelFormatter, costsChartTooltipText, getColorVariants, menuItemDefinitionOptions } from "./utils/highCharts";
+import { costAsPercentOfGdp, gdpReferenceYear } from "./utils/formatters";
+import { CostBasis } from "~/types/unitTypes";
+
+const props = defineProps<{
+  basis: CostBasis
+}>();
+
+const appStore = useAppStore();
+let chart: Highcharts.Chart;
+const seriesData = ref<Highcharts.SeriesColumnOptions[]>([]); // This only exists for testing purposes
+const chartContainer = ref<HTMLElement | null>(null);
+const chartParentEl = computed(() => chartContainer.value?.parentElement);
+
+// A range of colors per column, to be used for the breakdowns within each column.
+const columnColors = computed((): string[][] => {
+  return appStore.totalCost?.children?.map((cost, i) => {
+    const columnBaseColor = colorBlindSafeColors[i].hex;
+    const numberOfColorVariants = Math.max(cost.children?.length || 1);
+    return getColorVariants(columnBaseColor, numberOfColorVariants);
```

Comment:
Minor performance saving, but you could just calculate these colour variants once in the util, since they're always used in conjunction with the colour blindness safe colours, rather than tying it to the component load.